fendermint/vm/interpreter/src/fvm/state/query.rs

```diff
@@ -154,7 +137,7 @@ where
     pub fn call(&self, mut msg: FvmMessage, use_cache: bool) -> anyhow::Result<ApplyRet> {
         // If the sequence is zero, treat it as a signal to use whatever is in the state.
         if msg.sequence.is_zero() {
-            if let Some((_, state)) = self.actor_state(&msg.from)? {
+            if let Some((_, state)) = self.actor_state(use_cache, &msg.from)? {
```

Comment:
It this was moved inside the with_exec_state then it wouldn’t have to initialise the execution state twice.

Reply:
yeah, @aakoshh, I removed the `actor_state` call and directly used the `exec_state`.